### DISCUSSION TITLE
Focus mode activation not working

### DISCUSSION THREAD

**Discussion by bbbarron** [ttscoff/bunch-docs#196] [General]
I'm stumped. I'm a new user of Bunch, and I'm incredibly excited about using it, but I have been unable to get a bunch to activate a focus mode – either pre-existing (like Do Not Disturb) or new ones I have created. Outside the Bunch app I am able to start and stop Focus modes, but not inside the Bunch. This is true for the example Bunches as well. I have tried on my both my M1 MacBook Pro and my M1 iMac. I have culled the support forums and Google, but no one else has mentioned this problem. Is anyone else experiencing this? What could I be doing wrong?

**Comment by manuelhubacher** [comment]
I’ve run into the same problem after updating to Monterey. It seems like Focus Modes aren’t supported (yet; see #167). 

As I’m not using that many Focus Modes I’ve created some simple shortcuts in the Shortcut app for the time being that activate a focus (e.g. do not disturb) and one that turns off the active focus mode.

<img width="800" alt="shortcut to activate dnd" src="https://user-images.githubusercontent.com/17408325/152678425-7adb7494-22ac-411c-8bcb-9f743de21a17.png">

<img width="800" alt="shortcut to turn off focus modes" src="https://user-images.githubusercontent.com/17408325/152678550-ec325f54-6338-4d5a-90af-54912d69c04a.png">

(The shortcut above seems to turn off *all* focus modes and not just ‘do not disturb’ … 🤷🏻‍♂️)

I can run the shortcuts in my bunches by using the command line utility for running shortcuts (see [Running Shell Scripts in Bunch](https://bunchapp.co/docs/bunch-files/scripts/shell-scripts/)). So I add the following lines if I’d like to turn ‘do not disturb’ on when activating the bunch and turning it off when closing the bunch :

```
$ /usr/bin/shortcuts run "dnd"
!$ /usr/bin/shortcuts run "Focus off"
```

(The second line uses the ‘Run On Close’ feature, see [Run On Close](https://bunchapp.co/docs/bunch-files/run-on-close/).

**Reply by ttscoff** [reply]
This is an excellent tip, thanks @manuelhubacher. I haven't yet looked into what's required to alter Focus Mode states on macOS 12. It's always been a convoluted process of overwriting PLIST data in-memory that's a bit hacky, not sure if they made it any more accessible now.

**Reply by bbbarron** [reply]
Thank-you @manuelhubacher. I was beginning to think I was having a personal issue as no one else reported a problem. You workaround is a good solution. Thanks. And thank-you Brett for registering the problem

**Reply by drewkerr** [reply]
Regarding convoluted processes for altering Focus Mode, I [posted this on Automators](https://talk.automators.fm/t/get-current-focus-mode-via-script/12423/8), which might provide a starting point.
